### PR TITLE
Kill subcomponents

### PR DESCRIPTION
**Pull request by michsa**
Branched from #19 

* Remove all subcomponents: convert subcomponent variants (Banner, Box, Button, Tabs) into flag variants, and actual subcomponents (Modal, Form) into separate exports
* Remove alias props  (`p`/`px`/`py` on Box and `m` on Divider) - the long forms (`padding`/`margin`) still work
* Remove Button `tertiary` variant - the dark button styling is now the `secondary` variant (which was previously the same as the base button)
* Update storybook from 5.2.6 to 5.3.12

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+6792 -3810)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against da05dcb47630349cb21d791e4b6dda7a6891d33f
</p>

File: src/Banner.js [hunk 1]
```diff
@@ -1,51 +1,50 @@
...
+    justifyContent="center"
+    css={[
+      { padding: theme.spaces.xs },
+      styleFromVariant(getVariant(props, variants, 'regular')),
+    ]}
+    gap="xs"
+    {...props}
+  >
+    <Icon
+      icon={icon || getVariant(props, variants, 'regular').icon}
```
Comment (daedalus28) [inline src/Banner.js]: We should cache the variant in a local variable so we don’t run it twice if there’s an icon

File: src/utils.js [hunk 2]
```diff
@@ -27,3 +27,11 @@ export let findKeys = _.curry((predicate, data) =>
 export let coalesce = _.find(F.isNotNil)
 
 export let optionsFromArray = F.mapIndexed((label, value) => ({ label, value }))
+
+export let getVariants = (props, variants, defaultKey) =>
```
Comment (daedalus28) [inline src/utils.js]: This could use a unit test or at least a comment documenting its usage

Comment (michsa) [reply src/utils.js]: commented - I'd like to add a unit test at some point but now is probably not the best time

**Pull request merged**